### ISSUE TITLE
Getting NaN values when running nc_train.py

### ISSUE THREAD

**Issue by SivaHitesh047** [zaccharieramzi/fastmri-reproducible-benchmark#180]
Hi, 
I was trying to run the nc_train.py for the default parameters on the fastMRI single coil knee dataset and am getting “nan” values for all the metrics( loss, mean squared error, keras psnr, keras ssim). Can you please help me out and tell me what’s the problem?
<img width="1666" alt="Screenshot 2022-11-24 at 10 10 23 PM" src="https://user-images.githubusercontent.com/56303250/203833088-4e46d5f0-92e8-4184-af75-7845cba218af.png">

**Comment by chaithyagr** [comment]
Can you please tell what is the exact reproducible code you ran?
Also, please let us know the exact environment you used (tensorflow version etc).

**Comment by chaithyagr** [comment]
To me, on seeing your output, it is quite bizarre to have that high loss initially... Are you using density compensation? 
How did you set up the data?
From what I remember you need to convert the fastmri dataset into tfrecords to have preprocessed data (kspace data, smaps, density compensation, etc).

**Comment by SivaHitesh047** [comment]
I ran the _nc_train.py_ from the _training_scripts_ folder keeping all the parameters as it is. I used **_python -m nc_train_** command to run the file. I am not using density compensation. 

The dataset I used for train set are the .h5 files.

Here are some of the module versions I used in the environment.

tfkbnufft - 0.2.4
tensorflow -2.8.0                           
tensorflow-gpu - 2.4.1                            
tensorflow-nufft - 0.7.3  
scikit-image - 0.19.3
keras - 2.8.0
keras-preprocessing - 1.1.2

**Comment by chaithyagr** [comment]
Thank you for the info:

> The dataset I used for train set are the .h5 files.

Do you mean from fastMRI dataset? I am afraid that is not right. See here how the dataset needs to be organized: https://github.com/zaccharieramzi/fastmri-reproducible-benchmark/blob/4a4ec09524437d11153fc5a525621783689bed38/fastmri_recon/models/utils/fourier.py#L280-L289
Ideally you will need to generate the tfrecords dataset using: https://github.com/zaccharieramzi/fastmri-reproducible-benchmark/blob/master/fastmri_recon/data/scripts/multicoil_nc_tf_records_generation.py 
(Assuming you want to run multicoil in 2D)

> I am not using density compensation.
I Highly recommend you use density compensation. You can have scaling issues otherwise which can affect the convergence. See [the paper](https://hal.inria.fr/hal-03188997/document) Sec III D where the importance of it, particularly for non-Cartesian MRI is discussed.
By default we do use density compensation.

**Comment by SivaHitesh047** [comment]
Hey, thank you so much chaithya! It's working now. I was trying out the single coil only.

**Comment by SivaHitesh047** [comment]
I was wondering about adding k-space network (dual layers) to the NCPDNet (similar to how it is in the PDNet). Can you please suggest me on what to do? Also, is there any reason why you didn’t implement dual layers in NCPDNet.

**Comment by chaithyagr** [comment]
Usually k-space doesnt exactly satisfy the same properties of equivariance in image domain and hence such a direct network would not exactly help the cause in my opinion. However, if u still want to do the same, see that NCPDNet is derived from `CrossDomainNet`, which is exactly network in k-space then in image space.

What you would need to do is to modify here:

https://github.com/zaccharieramzi/fastmri-reproducible-benchmark/blob/4a4ec09524437d11153fc5a525621783689bed38/fastmri_recon/models/subclassed_models/ncpdnet.py#L75-L81


Note that currently we only have data consistency here. I would recommend you to have this, and in addition add your k-space network.
Do let us know if you see any improvements over the baseline, and after benchmarking, we would not mind adding it as a feature.

**Comment by zaccharieramzi** [comment]
I am going to close this issue since it seems resolved for now (at least the original nan problem).
Thanks @chaithyagr for handling it.
